lib/src/git.rs

```diff
@@ -1080,20 +1056,37 @@ pub fn reset_head(mut_repo: &mut MutableRepo, wc_commit: &Commit) -> Result<(),
         // change, otherwise use default stat.
         let stat = old_index
             .as_ref()
-            .and_then(|old_index| old_index.entry_by_path_and_stage(path, Stage::Unconflicted))
+            .and_then(|old_index| old_index.entry_by_path_and_stage(path, stage))
             .filter(|entry| entry.id.as_bytes() == id && entry.mode == mode)
             .map(|entry| entry.stat)
             .unwrap_or_default();
 
-        // It is safe to push the entry because we ensure that we only add each path
-        // once, and we sort the entries after we finish adding them.
+        // It is safe to push the entry because we ensure that we only add each path to
+        // a stage once, and we sort the entries after we finish adding them.
         index.dangerously_push_entry(
             stat,
             gix::ObjectId::from_bytes_or_panic(id),
-            gix::index::entry::Flags::from_stage(Stage::Unconflicted),
+            gix::index::entry::Flags::from_stage(stage),
             mode,
             path,
         );
+    };
+
+    // Use the merged parent tree as the Git index, allowing `git diff` to show the
+    // same changes as `jj diff`. If the merged parent tree has 2-sided conflicts,
+    // then the Git index will also be conflicted.
+    for (path, entry) in wc_commit.parent_tree(mut_repo)?.entries() {
+        let entry = entry?.simplify();
+        if let [left, base, right] = entry.as_slice() {
+            // 2-sided conflicts can be represented in the Git index
+            push_index_entry(&path, left, Stage::Ours);
+            push_index_entry(&path, base, Stage::Base);
+            push_index_entry(&path, right, Stage::Theirs);
+        } else {
+            // We can't represent many-sided conflicts in the Git index, so just add the
+            // first side as staged even if there are conflicts.
+            push_index_entry(&path, entry.first(), Stage::Unconflicted);
```

Comment:
Some IDEs (e.g. IntelliJ IDEA) use the index for their conflict resolution interfaces, and they ignore the materialized markers entirely. So if we put the first 3 as the conflict, then the user might accidentally resolve it using IntelliJ, which would lead to an incorrect resolution. So I think it's safer that we just don't add it as a conflict at all, since the user might think it's safe to use Git-based tools to resolve it if we do.